src/Symfony/Component/Console/Exception/NamespaceNotFoundException.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Exception;
+
+/**
+ * Represents an incorrect namespace typed in the console.
+ *
+ * @author Pierre du Plessis <pdples@gmail.com>
+ *
+ * @deprecated This class won't extend CommandNotFoundException in Symfony 5
+ */
+class NamespaceNotFoundException extends CommandNotFoundException
+{
+}
```

Comment:
This addition is not strictly needed, right? Not sure it's useful, and it involve a BC break with no upgrade path in the future (stop extending `CommandNotFoundException`, new unhandled exceptions in userland).

Reply:
No it's not strictly needed, but I needed a way to distinguish between not finding a namespace and not finding a command to prevent incorrect suggestions when typing an incorrect namespace. Keeping this class extending from `CommandNotFound` seemed weird as they are for different use cases, which is why I added the deprecation